kong/db/schema/init.lua

```diff
@@ -1728,26 +1766,10 @@ function Schema:process_auto_fields(data, context, nulls, opts)
               refs = { [key] = value }
             end
 
-            local deref, err = kong.vault.get(value)
-            if deref then
-              value = deref
-
-            else
-              if err then
-                kong.log.warn("unable to resolve reference ", value, " (", err, ")")
-              else
-                kong.log.warn("unable to resolve reference ", value)
-              end
-
-              value = nil
-            end
+            value = resolve_reference(value)
 
-          elseif prev_refs and prev_refs[key] then
-            if refs then
-              refs[key] = prev_refs[key]
-            else
-              refs = { [key] = prev_refs[key] }
-            end
+          else
+            refs = fill_prev_refs(prev_refs, refs, key)
```

Comment:
This deserves a comment - What's going on here?  I understand we're iterating over the fields and have determined that a value is a reference, so we resolve it.  But if it is not a value, what does `fill_prev_refs` do?  Previous to what?

We're in `process_auto_fields`, but apparently we're resolving references.  Why are the two functions conflated?  It would be clearer if we'd have a separate pass over the configuration to resolve the references - Fewer conditionals, less skipping between responsibilities.

Reply:
The `previous` thing I agree needs some commenting. This is the source of it:
https://github.com/Kong/kong/pull/9234

Reply:
One way to perhaps solve this is to just create $refs, and then call `vault.update`. But then `vault.update` would also perhaps need to actually check beyond cache. That would be the two pass thing, but I still feel that it might not be in right order as process auto fields does other stuff too. It might work, but might get tricky too.